tests/Feature/Proxy/FileStreamProxyTest.php

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /** @test */
+    public function testDetach()
+    {
+        $this->assertNotEmpty($this->fileStreamProxy->getSize());
+        $this->fileStreamProxy->detach();
+        $this->assertEmpty($this->fileStreamProxy->getSize());
+    }
+
+    /** @test */
+    public function testGetSize()
+    {
+        $this->assertEquals(25, $this->fileStreamProxy->getSize());
+    }
+
+    /** @test */
+    public function testPositionFunctions()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->fileStreamProxy->seek(9);
+        $this->assertEquals(9, $this->fileStreamProxy->tell());
+
+        $this->fileStreamProxy->rewind();
+        $this->assertEquals(0, $this->fileStreamProxy->tell());
+
+        $this->assertFalse($this->fileStreamProxy->eof());
+        $this->fileStreamProxy->seek(0, SEEK_END);
+        $this->fileStreamProxy->read(100);
+        $this->assertTrue($this->fileStreamProxy->eof());
+    }
+
+    /** @test */
+    public function testIsSeekable()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->assertTrue($this->fileStreamProxy->isSeekable());
+        $this->fileStreamProxy->seek(5);
+        $this->assertEquals(5, $this->fileStreamProxy->tell());
+
+        $this->fileStreamProxy->close();
+        $this->assertFalse($this->fileStreamProxy->isSeekable());
+        $this->expectException(RuntimeException::class);
+        $this->fileStreamProxy->seek(7);
+    }
+
+    /** @test */
+    public function testIsWritable()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->assertTrue($this->fileStreamProxy->isWritable());
+        $this->fileStreamProxy->write('This is new stream data written to the stream.');
+        $this->assertEquals(
+            'This is new stream data written to the stream.',
+            $this->fileStreamProxy->__toString()
+        );
+
+        $this->fileStreamProxy->close();
+        $this->assertFalse($this->fileStreamProxy->isWritable());
+        $this->expectException(RuntimeException::class);
+        $this->fileStreamProxy->write('This will trigger an exception.');
+    }
+
+    /** @test */
+    public function testIsReadable()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->assertTrue($this->fileStreamProxy->isReadable());
+        $this->assertEquals('This is', $this->fileStreamProxy->read(7));
+
+        $this->fileStreamProxy->close();
+        $this->assertFalse($this->fileStreamProxy->isReadable());
+        $this->expectException(RuntimeException::class);
+        $this->fileStreamProxy->read(4);
+    }
+
+    /** @test */
+    public function testGetContents()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->assertEquals('This is some stream data.', $this->fileStreamProxy->getContents());
+    }
+
+    /** @test */
+    public function testMetaDeta()
+    {
+        $this->fileStreamProxy->rewind();
+        $this->assertEquals(
+            'php://temp',
```

Comment:
You're now incorrectly assuming that a `FileStreamProxy` _always_ reads the data from temporary memory. If we were to use another `Stream` class in the proxy, it may have the uri set to the actual uri it's reading from (in this case: `https://api/v1/files/file-stream-id-1`). You can assert that the uri metadata is not empty, but I don't think there's any reasonable other assertion you can do.